main.js

```diff
@@ -63,64 +24,107 @@ class Gameboard {
-      } 
+      if (count === 3) {
+        this.gameOver = true
+        return this.gameOver
+      }
+    }
+  }
+
+  gameDrawn(selection1, selection2) {
+    if ((selection1 + selection2) === 9) {
+      this.gameOver = true
+      return this.gameOver
     }
   }
+}
 
-  // Tried to refactor out some part of the gamePlay function to make it cleaner but did not seem to work correctly
+//  Player Class
+
+class Player {
+  constructor(name, shape) {
+  this.selectedCells = []
+  this.name = name
+  this.shape = shape
+  }
 
-  // checkWinner(value) {
-  //   if (value == this.currentPlayer.name) {
-  //     return instructions.innerHTML = `Congratulations!!! ${this.currentPlayer.name}, You Win`;
-  //   } else if (value == 'draw') {
-  //     return instructions.innerHTML = `What a bore, Hit Restart`;
-  //   }
-  // }
-  
+  recordOption(index) {
+    this.selectedCells.push(index)
+  }
 }
 
-const game = new Gameboard();
+const player1 = new Player(prompt('Enter Your Name, Player 1'), 'X')
+const player2 = new Player(prompt('Enter Your Name, Player 2'), 'O')
 
-// console.log(game.endGame);
+let currentPlayer = player1
 
-let board = document.querySelector(".row");
+const layout = document.querySelector(".row");
 
-game.board.forEach((cell, index) => {
-  let div = document.createElement("div");
-  div.innerHTML = cell;
-  div.id = index;
-  div.classList = "col-md-4";
-  board.appendChild(div);
-});
+const gameBoard = new Board()
+
+gameBoard.options.forEach((option, index) => {
+  let div = document.createElement("div")
+  div.innerHTML = option
+  div.id = index
+  div.className = "col-md-4"
+  layout.appendChild(div)
+})
 
 let instructions = document.querySelector(".instructions");
-instructions.innerHTML = `It's your turn ${game.currentPlayer.name}, your shape is ${game.currentPlayer.shape}`
 
+instructions.innerHTML = `It's your turn ${currentPlayer.name}, your shape is ${currentPlayer.shape}`
 
-let cells = document.querySelectorAll(".col-md-4");
+const swapTurn = (current, first, second) => {
+  return current = (current === first) ? second : first
+}
+
+const cells = document.querySelectorAll(".col-md-4");
 
 cells.forEach((cell, index) => {
-  cell.addEventListener("click", () =>
-    game.gamePlay(game.currentPlayer.shape, index)
-  );
+  cell.addEventListener("click", () => {
+    gamePlay(cell, index)
+  });
 });
 
-const button = document.querySelector('button');
+const gamePlay = (cell, index) => {
+  if (!gameBoard.gameOver) {
+  if (cell.innerHTML === '') {
+    currentPlayer.recordOption(index)
+    gameBoard.optionChosen(index, currentPlayer.shape)
+    cell.innerHTML = currentPlayer.shape
+    if (gameBoard.gameWon(currentPlayer.selectedCells)) {
+      instructions.innerHTML = `Congratulations!!! ${currentPlayer.name}, You Win. Start New Game?`
+      return instructions.innerHTML      
+    } else if (gameBoard.gameDrawn(player1.selectedCells.length, player2.selectedCells.length)) {
```

Comment:
[Take a look at the no-else-return](https://eslint.org/docs/rules/no-else-return)